proxy/coco/src/lib.rs

```diff
@@ -55,3 +57,34 @@ pub use source::{
     tree, Blob, BlobContent, Branch, Commit, CommitHeader, Info, ObjectType, Person, Revision,
     Revisions, Tag, Tree, TreeEntry,
 };
+
+pub mod user;
+
+use librad::keys;
+use librad::net::discovery;
+use librad::net::peer::PeerConfig;
+use std::net::SocketAddr;
+
+/// Constructs a [`Peer`] and [`State`] pair from a [`PeerConfig`].
+///
+/// # Errors
+///
+/// * peer construction from config fails.
+/// * accept on the peer fails.
+pub async fn into_peer_state<I>(
+    config: PeerConfig<discovery::Static<I, SocketAddr>, keys::SecretKey>,
+    signer: librad::signer::BoxedSigner,
+    store: kv::Store,
+) -> Result<(Peer, Lock), Error>
+where
+    I: Iterator<Item = (PeerId, SocketAddr)> + Send + 'static,
+{
+    let peer = config.try_into_peer().await?;
+    let (api, run_loop) = peer.accept()?;
+
+    let state = State::new(api, signer);
+    let state = state::Lock::from(state);
+    let peer = Peer::new(run_loop, state.clone(), store);
+
+    Ok((peer, state))
+}
```

Comment:
This API is a bit weird to me, maybe it could work more like this:
```
let peer = Peer::from_config(config, signer, store)?; // this is basically implemented as the above function
let state = peer.state.clone();
```
?

Reply:
Or even better, could this be `Peer::new`?

Reply:
#890